db-scheduler-boot-starter/src/main/java/com/github/kagkarlsson/scheduler/boot/config/shutdown/ContextClosedStopper.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) Gustav Karlsson
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.kagkarlsson.scheduler.boot.config.shutdown;
+
+import com.github.kagkarlsson.scheduler.Scheduler;
+import org.springframework.context.event.ContextClosedEvent;
+import org.springframework.context.event.EventListener;
+
+public class ContextClosedStopper extends AbstractSchedulerStopper {
+  public ContextClosedStopper(Scheduler scheduler) {
+    super(scheduler);
+  }
+
+  @EventListener(ContextClosedEvent.class)
+  public void whenContextIsStopped() {
+    doStop();
+  }
+}
```

Comment:
Do we know when this event fires? Is it after all beans have been "destroyed"? Are we risking that some of the dependencies of db-scheduler is shutdown before the scheduler? 🤔